xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/transformation/macro/MacroTransformation.java

```diff
@@ -354,4 +358,69 @@ public void setMaxRecursions(int maxRecursions)
     {
         this.maxRecursions = maxRecursions;
     }
+
+    @Override
+    public void prepare(Block block)
+    {
+        Syntax syntax = null;
+
+        // Find the current syntax
+        MetaDataBlock metaDataBlock =
+            block.getFirstBlock(new MetadataBlockMatcher(MetaData.SYNTAX), Axes.ANCESTOR_OR_SELF);
+        if (metaDataBlock != null) {
+            syntax = (Syntax) metaDataBlock.getMetaData().getMetaData(MetaData.SYNTAX);
+        }
```

Comment:
A very similar logic exists in `org.xwiki.rendering.internal.macro.DefaultMacroContentParser#getCurrentSyntax` - wouldn't it make sense to use that method or to refactor the code to avoid the duplication?

Reply:
The main point of `DefaultMacroContentParser#getCurrentSyntax` is to find the syntax in the current context while here all we want is to find the syntax of a given block, which is `#getFirstBlock`'s job (which, yes, is also used by `DefaultMacroContentParser#getCurrentSyntax`). I definitely don't want to create a fake `MacroTransformationContext` just to reuse `getCurrentSyntax` but I guess we could image a `getSyntax(Block)` helper somewhere to avoid having to remember the parameters to pass to `#getFirstBlock`.

Reply:
After much thought on what was the less bad location and how generic it should be, I just committed a `Block#getSyntaxMetadata`.

Reply:
WDYT @michitux @surli (and others) ?

Reply:
Hm, feels a bit too specific in such a generic location. Maybe it could be a method `<T> T getMetaDataOnSelfOrAncestor(String metaData)`?

Reply:
I tough about this but if we do something too generic then there is not much point compared to directly using `block.getFirstBlock(new MetadataBlockMatcher(MetaData.SYNTAX), Axes.ANCESTOR_OR_SELF);` which you basically reproduced in the name of the method and its parameter.